### PR TITLE
New api module set to replace the hspy module

### PR DESCRIPTION
**Pull request by francisco-dlp**
This PR intends to implement the changes proposed in #494. It is still work in progress but ready for discussion.

TODO
--------

* Promote the usage of `import hyperspy.api as hs` in
    * [x] Docstrings
    * [x] User Guide
    * [x] Demos
* Rename/move:
    * [x] `hspy.components` -> `api.model_components` 
    * [x] `hspy.utils.*`-> `api.*`
    * [x] `hspy.utils.markers` -> `api.plot.markers`
    * [x] `hspy.utils.example_signals` -> `api.datasets.example_signals`
    * [x] Write automatic translator to run on the remaining code, User Guide, docstrings and demos and make it available to the users.

(Duplicate from #566 which I sent to the wrong branch)

### CODE REVIEW

**Comment by francisco-dlp** [comment]
This should be ready now. Note that it incorporate #564.

**Comment by francisco-dlp** [comment]
Ready for review

File: doc/user_guide/model.rst [hunk 1]
```diff
@@ -63,7 +63,7 @@ parameters for spectroscopy than the one that ships with HyperSpy:
 
 .. code-block:: python
 
-    >>> g = hs.components.Expression(
+    >>> g = hs.model.components.Expression(
```
Comment (to266) [inline doc/user_guide/model.rst]: I thought it's supposed to be `model_components`, not `model.components` ?

Comment (francisco-dlp) [inline doc/user_guide/model.rst]: Good that you raise this: I thought that, although at present ``components`` are alone in ``model``, the arrival of SAMFire in HyperSpy 0.9 will change this, making it worth it to have a ``model`` submodule.

File: hyperspy/api.py [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+
+The :mod:`~hyperspy.api` package contains the following submodules/packages:
+
+    :mod:`~hyperspy.api.signals`
+        `Signal` classes which are the core of HyperSpy. Use this modules to
+        create `Signal` instances manually from numpy arrays. Note that to
+        load data from supported file formats is more convenient to use the
+        `load` function.
+    :mod:`~hyperspy.api.model_components`
```
Comment (to266) [inline hyperspy/api.py]: needs changing if we have `model` module

File: hyperspy/api.py [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+        Components that can be used to create a model for curve fitting.
+    :mod:`~hyperspy.api.eds`
+        Functions for energy dispersive X-rays data analysis.
+    :mod:`~hyperspy.api.material`
+        Useful functions for materials properties and elements database that
+        includes physical properties and X-rays and EELS energies.
+    :mod:`~hyperspy.api.plot`
+        Plotting functions that operate on multiple signals.
+    :mod:`~hyperspy.api.example_signals`
```
Comment (to266) [inline hyperspy/api.py]: `datasets`

File: hyperspy/drawing/utils.py [hunk 4]
```diff
@@ -211,13 +211,13 @@ def plot_signals(signal_list, sync=True, navigator="auto",
...
     >>> s_ll = hs.load("lowloss.dm3")
-    >>> hs.utils.plot.plot_signals([s_cl, s_ll])
+    >>> hs.plot.plot_signals([s_cl, s_ll])
 
     Specifying the navigator:
 
     >>> s_cl = hs.load("coreloss.dm3")
     >>> s_ll = hs.load("lowloss.dm3")
-    >>> hs.utils.plot_signals([s_cl, s_ll], navigator="slider")
+    >>> hs.plot_signals([s_cl, s_ll], navigator="slider")
```
Comment (to266) [inline hyperspy/drawing/utils.py]: `hs.plot.plot_signals`

Comment (francisco-dlp) [inline hyperspy/drawing/utils.py]: All fixed, thanks!

**Pull request merged**